src/notification/toast.component.ts

```diff
@@ -49,7 +50,17 @@ export class Toast extends BaseNotification implements OnInit {
 	 *
 	 * `type` can be one of `"error"`, `"info"`, `"info-square"`, `"warning"`, `"warning-alt"`, or `"success"`
 	 */
-	@Input() notificationObj: ToastContent;
+	@Input() set notificationObj(obj: ToastContent) {
+		if (obj.closeLabel) {
+			obj.closeLabel = of(obj.closeLabel);
```

Comment:
You would require the `| async` in the template where it's using the label...but tbh i tried to understand why it was introduced that way, and i can't get it...it seems and attempt to have closeLabel as observable as well as string, but in this way if i set an observable in closeLabel, it will only be an `Observable<Observable<string>>`

Reply:
Good catch, thanks!

When we pass a non-observable parameter in `of`. It creates a observable and emits the non-observable value. When we pass an observable as a parameter in `of`, it subscribes to it and emits the value. 

So by using `of` operator, we don't have to check the content if closeLabel is an observable or a string.

Reply:
You're right, scratch my last statement, it wraps the observable within another observable, so subscribing to it would emit an observable.

So from what I'm seeing in the v10 code, we do NOT support observable for `closeLabel`. We have the `of` in place to convert the user passed `closeLabel` value to an observable which is unwrapped by the `async` pipe.

We DO use the observable initially for defaultNotificationObj which fetches translations and returns a BehaviorSubject:
```ts
// Returns a STRING or an Observable
closeLabel: this.i18n.get("NOTIFICATION.CLOSE_BUTTON"),
```

Looking at the [PR that made this change](https://github.com/carbon-design-system/carbon-components-angular/pull/299), `closeLabel` initially had a type of `string` which was then set to `any` .

We can change the setter to the following to support observables, but the question is do we really want to support this?
```ts
	set notificationObj(obj: NotificationContent) {
		if (obj.closeLabel && !isObservable(obj.closeLabel)) {
			obj.closeLabel = of(obj.closeLabel);
		}
		this._notificationObj = Object.assign({}, this.defaultNotificationObj, obj);
	}
```

We can set the type of `closeLabel` to string and set the initial translated value in BaseNotification constructor.

@zvonimirfras thoughts?

Reply:
any update on this @Akshat55 ?

Reply:
Hey @Licen-it, this is planned for end of this week. I'll be picking it up sometime tomorrow. We plan on removing `async` pipe altogether for notification and setting the type in the interface to string. Just need to run a few tests before making a these changes.

Reply:
Decided to check for observables, if user passes in an observable we don't use the `of` operator. We don't have to manually subscribe then unsubscribe if we use the `async` pipe. Depending on the usage of the translate service, we may remove it in distant future.... I don't see a reason to use it over ngx-translate or other libs.